x-pack/plugin/fleet/src/main/java/org/elasticsearch/xpack/fleet/action/TransportDeleteSecretAction.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.fleet.action;
+
+import org.elasticsearch.ResourceNotFoundException;
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.DocWriteResponse.Result;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.HandledTransportAction;
+import org.elasticsearch.action.support.WriteRequest;
+import org.elasticsearch.client.internal.Client;
+import org.elasticsearch.client.internal.OriginSettingClient;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.transport.TransportService;
+
+import static org.elasticsearch.xpack.core.ClientHelper.FLEET_ORIGIN;
+import static org.elasticsearch.xpack.fleet.Fleet.FLEET_SECRETS_INDEX_NAME;
+
+public class TransportDeleteSecretAction extends HandledTransportAction<DeleteSecretRequest, DeleteSecretResponse> {
+
+    private final Client client;
+
+    @Inject
+    public TransportDeleteSecretAction(TransportService transportService, ActionFilters actionFilters, Client client) {
+        super(DeleteSecretAction.NAME, transportService, actionFilters, DeleteSecretRequest::new);
+        this.client = new OriginSettingClient(client, FLEET_ORIGIN);
+    }
+
+    @Override
+    protected void doExecute(Task task, DeleteSecretRequest request, ActionListener<DeleteSecretResponse> listener) {
+        client.prepareDelete(FLEET_SECRETS_INDEX_NAME, request.id())
+            .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
```

Comment:
I am not sure why this is needed - we don't really need to do a refresh here because we are not searching over the fleet secrets index, we do a GET document request for the GET secret API and AFAIK GET document does not need a refresh to return the latest value.
(same comment for POST)